### DISCUSSION TITLE
How to compare values in columns with encoded values?

### DISCUSSION THREAD

**Discussion by IvanVakhatov** [moj-analytical-services/splink#964] [Q&A] [answered]
Hello everyone, I have several columns in my dataset: Address, CompanyName, Region, City and so on. And also 2 columns that contain encoded values:
![image](https://user-images.githubusercontent.com/48402122/209676581-767e3d24-ae91-4e77-b738-738f48976114.png)
(like "c" stands for "pharmacy, "g" for "warehouse" and so on).

I have to consider intersections between these values, so I tried to use **_splink.duckdb.duckdb_comparison_library.jaccard_at_thresholds()_** method, but it threw warning during the **_linker.estimate_u_using_random_sampling()_**:
![image](https://user-images.githubusercontent.com/48402122/209677146-215a11ce-0e75-4862-ab46-9f99d9e80465.png)
I also tried to use levenshtein_at_thresholds(), but this error has occurred again.

**_exact_match()_** method is not suitable, I guess, and also in a dataset of size 30,000, there are only 20 unique values in the "Customer_type" and "Company_type" fields, but intersections are very important and valuable for my task.

Please tell me if there are any ways / methods to solve this a problem.
Thanks!

**Comment by RossKen** [comment, marked as answer]
Hi @IvanVakhatov,

This will be occurring because there are some cases for your comparison which are not seen within your sample. e.g. no combination of "Customer types" with levenshtein = 2 (or whatever your metrics are).

Given your dataset is relatively small, you can just use the entire dataset to estimate u - `linker.estimate_u_using_random_sampling(target_rows = 30000)` 

If increasing your sample doesn't work, I would suggest changing your comparison definition. For example, if you are using `levenshtein_at_thresholds()`, given your column entries are only two letters long you won't have a levenshtein value ever equal to 3 or above.
